Pocket.Logger/Core/Formatter.cs

```diff
@@ -1,193 +1,312 @@
                     }
+                }
 
-                    sb.Replace(replacementTarget, formattedParam);
+                if (formattedParam is null)
+                {
+                    formattedParam = value.ToLogString();
                 }
 
-                argumentFormatters.Add(Format);
+                sb.Replace(replacementTarget, formattedParam);
             }
+
+            argumentFormatters.Add(Format);
         }
+    }
 
-        public IReadOnlyList<string> Tokens => tokens;
+    public IReadOnlyList<string> Tokens => tokens;
 
-        public FormatterResult Format(
-            IReadOnlyList<object> args,
-            IList<(string Name, object Value)> knownProperties)
+    public FormatterResult Format(
+        IReadOnlyList<object> args,
+        IList<(string Name, object Value)> knownProperties)
+    {
+        if (args is null)
         {
-            if (args == null)
-            {
-                args = new object[argumentFormatters.Count];
-            }
+            args = new object[argumentFormatters.Count];
+        }
 
-            var stringBuilder = new StringBuilder(template);
-            var result = new FormatterResult(stringBuilder);
+        var stringBuilder = new StringBuilder(template);
+        var result = new FormatterResult(stringBuilder);
 
-            for (var i = 0; i < Math.Min(argumentFormatters.Count, args.Count); i++)
-            {
-                var argument = args[i];
-                argumentFormatters[i](stringBuilder, argument);
-                result.Add(Tokens[i], argument);
-            }
+        for (var i = 0; i < Math.Min(argumentFormatters.Count, args.Count); i++)
+        {
+            var argument = args[i];
+            argumentFormatters[i](stringBuilder, argument);
+            result.Add(Tokens[i], argument);
+        }
 
-            if (args.Count > argumentFormatters.Count || knownProperties?.Count > 0)
-            {
-                stringBuilder.Append(" +[ ");
-                var first = true;
+        if (args.Count > argumentFormatters.Count || knownProperties?.Count > 0)
+        {
+            stringBuilder.Append(" +[ ");
+            var first = true;
 
-                if (args.Count > 0)
+            if (args.Count > 0)
+            {
+                for (var i = argumentFormatters.Count; i < args.Count; i++)
                 {
-                    for (var i = argumentFormatters.Count; i < args.Count; i++)
-                    {
-                        var argument = args[i];
-                        TryAppendComma();
-                        stringBuilder.Append(argument.ToLogString());
-                        result.Add($"arg{i}", argument);
-                    }
+                    var argument = args[i];
+                    TryAppendComma();
+
+                    stringBuilder.Append(argument.ToLogString());
+                    result.Add($"arg{i}", argument);
                 }
```

Comment:
On line 128 below we call `ToLogString` on `property` which is a tuple of type `(string Name, object Value)`. Could you change this so that we call `ToLogString` on `property.Value` instead (with `property.Name` being logged verbatim)?